### DISCUSSION TITLE
Patient informations

### DISCUSSION THREAD

**Discussion by testlorenzospa** [FirelyTeam/firely-net-sdk#2510] [Q&A]
Hello.

I am not very expert about FHIR and this library.
But I want to collect with this library some patient informations that I haven't found.

For example, for a patient I want to collect these data if existent

- **Telephone numbers (primary and secondary, e.g. home phone or cellular number)**
- **Email**
- health insurance card
- Tax ID code

How can I collect these data with this library?

**UPDATE**

Hello.

I have found how to get the two first information (Telephone numbers and Email but not the last two).

Thanks.

**UPDATE 2**

Hello again.

I have found how to get the last two informations (health insurance card and Tax ID code).
But now, I don't know how to get the right identifier (if there are more than one).
How can I identify the right identifier to get? Is there an enumerable to use for this?

Many thanks.

**Comment by mbaltus** [comment]
Thank you for updating us on your finds.
The Identifier has a field called 'system'. This is usually filled with a url(preferred) or uri to indicate the namespace for the identifier value. Those systems are often indicated in an Implementation Guide for the project/context that you need to work with. If your own organization issues identifier values, your organization can create such a system url to be used in the FHIR data.

Since the list of systems is not fixed and are often locally defined, there is no enumerable for them in the library. 

Edit to add: You can also add a general category to your identifiers by using the 'type' field. Again, there's no fixed list for those, so no enumerable, but you can find the current recommended list of types here: http://hl7.org/fhir/valueset-identifier-type.html.

**Reply by testlorenzospa** [reply]
Hello @mbaltus 

Thanks again for your reply.
Do you have some code to use for this verify?
My only scope is to do a verify for the identifiers like with the address for example (where the library have an enumerable and I can check a value from the bundle json to know if the address is the right address to use).

Thanks for your support.

**Reply by mbaltus** [reply]
Here's an example where I do a search for all Patients on a system - no search parameters used - and then from the result select only the Patients that have a specific system url filled in for (one of) the Identifier(s):

```csharp
FhirClient client = new FhirClient("https://server.fire.ly");

// Retrieving all Patients - note that this server will only send 10 per Bundle
Bundle result = await client.SearchAsync<Patient>();

// Getting the Patient resources from the Bundle first
List<Patient> patients = (List<Patient>) result.GetResources().Where(r => r is Patient);
// Selecting Patients that have an Identifier with a specific system url
List<Patient> selectedPats = patients.Where(p => p.Identifier.Exists(i => i.System == "https://github.com/synthetichealth/synthea")).ToList();
```

I could have added a search parameter to the Search in order to filter on the system url server side, but I thought you wanted to see how to manipulate the Bundle afterwards.

**Comment by testlorenzospa** [comment]
Hello @mbaltus 

Thanks.

Probably something is wrong in my bundle JSON string or there is other issue.

This is the example of a fixed blundle JSON string that at the moment I am reading in my code

`{ "resourceType": "Bundle", "id": "example-bundle", "type": "collection", "entry": [ { "resource": { "resourceType": "Patient", "id": "patient1", "identifier": [ { "type": { "coding": [ { "system": "http://terminology.hl7.org/CodeSystem/v2-0203", "code": "TAX", "display": "TAX ID number" } ], "text": "TAX ID number" }, "value": "ABCXYZ12L34M567N" } ], "name": [ { "given": [ "Mario" ],"family": "Rossi" } ], "gender": "male", "birthDate": "1980-01-01", "deceasedDateTime" : "2023-05-10T08:00:00Z", "maritalStatus": { "coding": [ { "system": "http://hl7.org/fhir/v3/MaritalStatus", "code": "M", "display": "Married" } ] }, "address": [ { "use": "home", "text": " Via Livorno, 20", "city": "058091", "state": "00RM", "postalCode": "00162", "country": "IT", "extension": [ { "url": "http://hl7.org/fhir/StructureDefinition/address-official", "valueBoolean": true } ] } ], "extension": [ { "url": "http://hl7.org/fhir/StructureDefinition/patient-citizenship", "valueString": "Italiana" }, { "url": "http://hl7.org/fhir/StructureDefinition/patient-birthPlace", "valueAddress": { "city": "Roma" } }, { "url": "http://hl7.org/fhir/StructureDefinition/patient-codeableBirthPlace", "valueString": "058091" }, { "url": "http://hl7.it/fhir/StructureDefinition/patient-qualification-it", "valueCodeableConcept": { "coding": [ { "system": "http://terminology.hl7.it/CodeSystem/istat-ctsi03", "code": "10000000", "display": "LICENZA ELEMENTARE" } ], "text": "LICENZA ELEMENTARE" } }, { "url": "http://hl7.it/fhir/StructureDefinition/patient-occupation-it", "valueCodeableConcept": { "coding": [ { "system": "http://terminology.hl7.it/CodeSystem/istat-professioni", "code": "1", "display": "Dirigenti ed equiparati delle università e degli enti di ricerca" } ], "text": "Dirigenti ed equiparati delle università e degli enti di ricerca"} } ] } } ] }`

In this example, for the section "identifier" I can see an identifier (with code "TAX") with a System and value.
For System, from this example I can see this url "http://terminology.hl7.org/CodeSystem/v2-0203".
For value, I can see "ABCXYZ12L34M567N".

From your code, if I well understand if the System is equals to an exact URL I can found patients.
But in this way, I still don't distinguish an identifier to another identifier if there are multiple System equals. Is it correct?
Is there the possibility to read the value "code" instead the System? Or System is the more standard way to intercept an identifier?

Another issue.
I have tried to read System from the identifiers but I haven't a value.
Below, how I read the System identifiers.

`foreach (var ident in pat_entry.Identifier)
    {
        Console.WriteLine("Patient's system: " + ident.System);
    }`
But the ident.System is blank.
Is there something wrong in this code?

Thanks again for your patience.

**Reply by mbaltus** [reply]
Identifiers have a namespace, which can be listed in the Identifier.System field. Identifiers can also have a type and the type then contains a code. Codes also do have a system: the code system the code comes from. So in your example, yes, there is a URL mentioned, but it is the Identifier.Type.Coding.System and not the Identifier.System. In other words: the URL belongs to the code, and not to the identifier value.

Most FHIR data will have an Identifier.System used when they want to be more specific in indicating where the identifier value comes from. The Identifer.Type is used for more generic classifications, like "this is a Tax ID" - regardless of what tax office or country has issued it.

Since you have the Identifier.Type in your data, you can change my code example to use that, instead of the Identifier.System:
```csharp
List<Patient> selectedPats = patients.Where(p => p.Identifier.Exists(i => i.Type.Coding.First().Code == "TAX")).ToList();

```
